src/basic/stat-util.c

```diff
@@ -151,6 +151,26 @@ int is_device_node(const char *path) {
         return verify_stat_at(AT_FDCWD, path, false, stat_verify_device_node, false);
 }
 
+int stat_verify_regular_or_block(const struct stat *st) {
```

Comment:
This seems to serve a quite specific use case and not something generic enough. Just call fstat() directly and do type check in repart?